source/agora/node/Registry.d

```diff
@@ -583,7 +617,9 @@ private SOA fromConfig (in ZoneConfig zone, Domain name) @safe
     return SOA(
         // mname, rname
         Domain.fromString(zone.primary),
-        Domain.fromString(zone.soa.email.value.replace('@', '.')),
+        (zone.soa.email.length > 0) ?
+            Domain.fromString(zone.soa.email.value.replace('@', '.'))
+            : Domain.init, // Secondary will get this from primary
```

Comment:
It will get the rest from primary too, so maybe we should just not be calling `fromConfig` for non-primary ?

Reply:
I see, non-primaries call this to initialize timing fields of their SOA as a backup value (in-case of they couldn't reach any primary to get actual SOA)

Reply:
I can either set default values for time values of `SOA` struct in `DNS` or keep this way of initializing it. `SOAConfig` already have default values which I wanted to utilize.
[This](https://github.com/mkykadir/agora/blob/a767a72dae19e96e6c423c70fe3223f99401de12/source/agora/node/Registry.d#L1026) is the reason why we need some default values.